src/pci/context_vars.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Functions to work with correlation ID.
+
+These should be moved into a library like GHGA-Service-Commons.
+"""
+from contextlib import asynccontextmanager
+from contextvars import ContextVar
+from typing import Any
+
+correlation_id_var: ContextVar[str] = ContextVar("correlation_id", default="")
+
+__all__ = [
+    "set_context_var",
+    "set_correlation_id",
+    "get_correlation_id",
+]
+
+
+@asynccontextmanager
+async def set_context_var(context_var: ContextVar, value: Any):
+    """An async context manager to simplify the use of ContextVars"""
+    token = context_var.set(value)
+    yield
+    context_var.reset(token)
+
+
+@asynccontextmanager
+async def set_correlation_id(correlation_id: str):
+    """Set the correlation ID."""
+    async with set_context_var(correlation_id_var, correlation_id):
+        yield
+
+
+def get_correlation_id() -> str:
```

Comment:
Maybe raise an exception if no correlation ID was set.

Reply:
It seems reasonable to assume that if you're using the function, then there should be a value and if no value is present, something has gone wrong. Updated in [eacba71](https://github.com/ghga-de/prototype-correlation-id/pull/1/commits/eacba71e1c0874313864fb3cef72cfc6f61d41e9)